### PR TITLE
feat: use podio collections in tracking digi/reco; rm unused interfaces (3rd attempt)

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This converts SiliconTrackerDigi (used for MPGDs too) and TrackerHitReconstruction to processing podio collections instead of individual hits. It adapts their factories to multifactories, and updates the plugins. It removes now unused algorithm interface definitions (IObjectProducer, etc) and the remaining duplicated and unused TruthTrackSeeding algorithm.

Follows #840 and #871.

### What kind of change does this PR introduce?
- [x] Bug fix (issue: remove unused algorithms, interfaces, and factories)
- [x] New feature (issue: move tracking digitization and reconstruction to podio collections)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No. This should lead to identical output as before..

### Does this PR change default behavior?
No.

### CODE REVIEW

**Comment by wdconinc** [comment]
- Brycecanyon: no changes https://github.com/eic/EICrecon/actions/runs/5977798800/job/16219015583?pr=872#step:10:72
- Craterlake: no changes https://github.com/eic/EICrecon/actions/runs/5977798800/job/16219015745?pr=872#step:10:72

File: src/factories/digi/SiliconTrackerDigi_factoryT.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// Copyright 2023, Wouter Deconinck
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
```
Comment (veprbl) [inline src/factories/digi/SiliconTrackerDigi_factoryT.h:3]:
```suggestion
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright (C) 2023 Wouter Deconinck
```

File: src/factories/tracking/TrackerHitReconstruction_factoryT.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
+// Copyright 2023, Wouter Deconinck
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
```
Comment (veprbl) [inline src/factories/tracking/TrackerHitReconstruction_factoryT.h:3]:
```suggestion
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright (C) 2023 Wouter Deconinck
```

**Review by veprbl — COMMENTED** [review]
No changes in output seen on CI

**Review by veprbl — APPROVED** [review]

**Pull request merged**